### DISCUSSION TITLE
ts-sdk your_coin.ts example why not FA transfer

### DISCUSSION THREAD

**Discussion by daorren** [aptos-labs/aptos-developer-discussions#247] [Questions] [answered]
### Discord user ID

0xdaorren

### Describe your question in detail.

I use examples/your_coin.ts in ts-sdk to create new coin in testnet
this is a transfer tx from alice to bob: https://explorer.aptoslabs.com/txn/1219219381/userTxnOverview?network=testnet
my question is, why this tx uses a coin_store rather than a primary_fungible_store

according to this line https://github.com/aptos-labs/aptos-core/blob/702abf46cf4937f130d0d257815a9efdf8c05d25/aptos-move/framework/aptos-framework/sources/coin.move#L875
when coin::deposit(which used by aptos_account::deposit), if coin_store not found on receiver's address, primary_fungible_store should be created and used
bob, in this case is new account for mooncoin, so there's no coinstore for mooncoin, and is supposed to use a primary_fungible_store. however, coinstore is used, which I can't understand

### What have you tried or looked at?  Or how can we reproduce the error?

code

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by kshitijc1506** [comment, marked as answer]
Hey, so you're on the right track. 

`aptos_account::transfer_coins` -> `deposit_coins` -> `is_account_registered` -> `register` -> `deposit`

Since the account wasn't registered for the Coin, and the [migrated Primary Fungible Store (PFS) **_doesn't_** exist for the account](https://github.com/aptos-labs/aptos-core/blob/702abf46cf4937f130d0d257815a9efdf8c05d25/aptos-move/framework/aptos-framework/sources/coin.move#L767), we register the account for the Coin Type. Therefore, the Coin Store **_does_** exist for the account.

**Reply by kshitijc1506** [reply]
Also, this is a bit of extra info if you're curious:

If you call directly into [coin::transfer](https://github.com/aptos-labs/aptos-core/blob/702abf46cf4937f130d0d257815a9efdf8c05d25/aptos-move/framework/aptos-framework/sources/coin.move#L1130), [this](https://github.com/aptos-labs/aptos-core/blob/702abf46cf4937f130d0d257815a9efdf8c05d25/aptos-move/framework/aptos-framework/sources/coin.move#L891) will not work since the FA Metadata object hasn't been paired to the Coin type. Also, if that passes, the [MigrationFlag](https://github.com/aptos-labs/aptos-core/blob/702abf46cf4937f130d0d257815a9efdf8c05d25/aptos-move/framework/aptos-framework/sources/coin.move#L211) resource also needs to be moved to the `to` [account's PFS](https://github.com/aptos-labs/aptos-core/blob/702abf46cf4937f130d0d257815a9efdf8c05d25/aptos-move/framework/aptos-framework/sources/coin.move#L691). 

This will be handled by Labs soon in `coin::deposit/withdraw`, so developers and users won't have to worry about it. However, it hasn't been implemented yet.
